pkg/common/worker_test.go

```diff
@@ -0,0 +1,23 @@
+package common_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/matrix-org/waterfall/pkg/common"
+)
+
+func BenchmarkWorker(b *testing.B) {
+	workerConfig := common.WorkerConfig[struct{}]{
+		Timeout:   2 * time.Second,
+		OnTimeout: func() {},
+		OnTask:    func(struct{}) {},
+	}
+	w := common.StartWorker(workerConfig)
+
+	for n := 0; n < b.N; n++ {
+		w.Send(struct{}{})
```

Comment:
This always returns an error, because we have to add an buffer for the channel in the WorkerConfig.

Reply:
Oh, good catch! 👍 Thanks!